src/main/java/baritone/pathing/movement/MovementHelper.java

```diff
@@ -88,61 +88,105 @@ static boolean canWalkThrough(BlockStateInterface bsi, int x, int y, int z) {
         return canWalkThrough(bsi, x, y, z, bsi.get0(x, y, z));
     }
 
+    static boolean canWalkThrough(CalculationContext context, int x, int y, int z, IBlockState state) {
+        return context.precomputedData.canWalkThrough(context.bsi, x, y, z, state);
+    }
+
+    static boolean canWalkThrough(CalculationContext context, int x, int y, int z) {
+        return context.precomputedData.canWalkThrough(context.bsi, x, y, z, context.get(x, y, z));
+    }
+
+
     static boolean canWalkThrough(BlockStateInterface bsi, int x, int y, int z, IBlockState state) {
+        Optional<Boolean> canWalkOn = canWalkThroughBlockState(state);
+        return canWalkOn.orElseGet(() -> canWalkThroughPosition(bsi, x, y, z, state));
```

Comment:
Let's avoid `orElseGet` in this case. It requires constructing that lambda `() -> canWalkThroughPosition(bsi, x, y, z, state)` which allocates on the heap because it's impure (meaning that it captures those variables bsi x y z state and allocates with them). Let's just do the old way (an if statement on `isPresent`).

Reply:
Sorry yeah, I had it like that originally, then I received complaints from intellij so thought this was what you are meant to do.